guides/source/active_record_basics.md

```diff
@@ -374,3 +374,86 @@ and to roll it back, `rake db:rollback`.
 Note that the above code is database-agnostic: it will run in MySQL,
 PostgreSQL, Oracle and others. You can learn more about migrations in the
 [Active Record Migrations guide](migrations.html).
+
+Enum
+----
+
+Active Record `enum` allows you declare an attribute where the values map to
+integers in the database, but can be queried by name.
+
+You can define the enum on a model calling the `enum` macro.
+
+```ruby
+class Post < ActiveRecord::Base
+  enum status: [ :active, :archived ]
+end
+```
+
+This will add `enum` on the column `status`, where the active is represented as 0
+and the archived is represented as 1.
+
+It's also possible to explicitly map the relation between attribute and
+database integer with a `Hash`:
+
+```ruby
+class Post < ActiveRecord::Base
+  enum status: { active: 0, archived: 1 }
+end
+```
+
+You can set the default value from the database declaration, like:
+
+```ruby
+create_table :posts do |t|
+  t.column :status, :integer, default: 0
+end
+```
+
+Good practice is to let the first declared status be the default.
+
+You can change between the mapped statuses with bang methods:
```

Comment:
What do you think about exposing bang and predicate methods together like:

> The `enum` macro will also give you access to bang and predicate methods to change
> or check the current value of the attribute:
>
> ~~~ruby
> post.active!
>
> post.archived? # => false
> post.active?   # => true
> ~~~

?